File: 0005_Longest Palindromic Substring/LongestPalindromicSubstring.py
class Solution:
  def longestPalindrome(self, s: str) -> str:
    n = len(s)
    def getLen(l, r):
      while l >= 0 and r < n and s[l] == s[r]:
        l -= 1
        r += 1
      return r - l - 1
    
    start = 0
    length = 0    
    for i in range(n):      
      cur = max(getLen(i, i),
                getLen(i, i + 1))
      if cur <= length: continue
      length = cur
      start = i - (cur - 1) // 2
    return s[start : start + length]

File: 0005_Longest Palindromic Substring/LongestPalindromicSubstring.cs
public class Solution {
   private int lo;
   private int max;

   public string LongestPalindrome(string s) {
       if(string.IsNullOrEmpty(s)) return "";
       if(s.Length == 1) return s;

       for(int i=0;i<s.Length-1;i++){
           LengthOfPalindrome(s, i, i);
           LengthOfPalindrome(s, i, i+1);           
       }

       return s.Substring(lo,max);
   }

   private void LengthOfPalindrome(string s, int i, int j)
   {
       while(i>=0 && j<s.Length && s[i] == s[j])
       {           
           i--;
           j++;           
       }

       if(j-i-1>max)
       {
           max = j-i-1;
           lo = i+1;
       }
   }
}
